### PR TITLE
Rewrite website infrastructure

### PR DESCRIPTION
**Pull request by Ampflower**
This includes edits to various pages to make it flow better, and adding Actions.

Tho the main focus is the package management and boilerplate reduction, along with cleaning up and minimizing the need for external assets.

### CODE REVIEW

File: src/routes/faithfulpride/+page.svelte.md [hunk 1]
```diff
@@ -3,49 +3,48 @@ import Badge from '$lib/components/BadgeRaw.svelte';
...
+        label="GitHub Downloads"
+        alt="GitHub Downloads (all assets, all releases)"/>
+<Shield href="https://git.pridecraft.gay/faithfulpride/commits"
+        base="github/commit-activity/t/pridecraft-studios/faithfulpride"
+        logo="github"
+        alt="GitHub Commit Activity"/>
+</aside>
 
-## Where else is this available?
+This is Faithful Pride. A LGBTQ+ Pride themed Resource Pack made by [Pridecraft Studios](https://git.pridecraft.gay/),
```
Comment (Ampflower) [inline src/routes/faithfulpride/+page.svelte.md:31]: I evidently didn't catch this before committing

File: src/routes/pridegui/+page.svelte.md
Comment (Ampflower) [inline src/routes/pridegui/+page.svelte.md:1]: Perhaps, give badge download buttons to the legacy version of the pack as well?

File: .gitignore
Comment (Ampflower) [inline .gitignore:1]: Probably should add lock and log files here. Unsure if there's other junk files to add in here.

File: .vscode/launch.json
Comment (Ampflower) [inline .vscode/launch.json:1]: This probably can be removed; it doesn't look like it'd function correctly.

File: README.md
Comment (Ampflower) [inline README.md:1]: Maybe just mention how to run it locally instead.

File: src/lib/components/Link.svelte
Comment (Ampflower) [inline src/lib/components/Link.svelte:1]: Probably better in boilerplate

File: src/lib/components/Popup.svelte [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+	export let id: string;
+	export let label: string;
+</script>
+
+<div class="popup">
+	<input {id} type="checkbox" hidden />
+	<label for={id}>
+		{label}
+		<ChevronRightIcon class="inline" />
```
Comment (Ampflower) [inline src/lib/components/Popup.svelte:12]: Perhaps investigate a better placement, and maybe not embedding the SVG into the HTML.

File: src/routes/+layout.js
Comment (Ampflower) [inline src/routes/+layout.js:1]: Should maybe be a typescript file.

File: src/routes/faithfulpride/+page.svelte.md [hunk 3]
```diff
@@ -3,49 +3,48 @@ import Badge from '$lib/components/BadgeRaw.svelte';
...
-## How can I contribute?
-You can help us translate Pridepack to more languages on
-[Crowdin](https://crowdin.com/project/pridepack/settings) [![Crowdin](https://badges.crowdin.net/pridepack/localized.svg)](https://crowdin.com/project/pridepack)
-all translations will be added to packs such as Faithful Pride and Pride Bones.
-You can also indirectly contribute by voting on polls on [the Discord](https://discord.pridecraft.gay), or [our Mastodon](https://tech.lgbt/@pridecraft)
+# Faithful Pride
 
+<aside class="shields">
+<ModrinthShield modid="faithfulpride"/>
+<CurseForgeShield modid="faithful-pride" type="texture-packs" projectid="831872"/>
```
Comment (Ampflower) [inline src/routes/faithfulpride/+page.svelte.md:19]: Update project ID when possible.

File: src/routes/frog/+page.svelte.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+<script lang="ts">
+let value = Math.floor(Math.random() * 999) + 1;
+</script>
+
 # You have been frogged!
+
 ![gay frog is not real and cant hurt you. gay frog:](https://github.com/Pridecraft-Studios/website/blob/main/static/assets/img/frog/frog.png?raw=true)
```
Comment (Ampflower) [inline src/routes/frog/+page.svelte.md:7]: This is not how to embed links; use `static` as if it was root; `/assets/img/frog/frog.png`

**Review by Ampflower — COMMENTED** [review]
*proceeds to self-review; finds several nitpicks*

File: src/routes/pridegui/+page.svelte.md
Comment (blryface) [reply src/routes/pridegui/+page.svelte.md:1]: ehhh we can have a single link to it on the project description

File: README.md
Comment (blryface) [reply README.md:1]: Skye removed that readme a few months back because I kept testing on prod 😭

File: src/routes/faithfulpride/+page.svelte.md (hunk 3 above)
Comment (blryface) [reply src/routes/faithfulpride/+page.svelte.md:19]: It kinda needs to exist first sob

Comment (Ampflower) [reply src/routes/faithfulpride/+page.svelte.md:19]: Fair enough; I'll just zero it then or make it intentionally fail?

**Comment by Ampflower** [comment]
Noticed some oversized PNG assets; considering what to do for those. I'd say probably JPEG since that gets crushed the least by proxies out of the current formats, but I don't know.

File: README.md
Comment (KTrain5169) [reply README.md:1]: stop testing on prod; cloudflare has staging deployments for a reason

Comment (Ampflower) [reply README.md:1]: Let's maybe not

File: pictures.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+All encoders are tuned to run at their slowest but highest quality possible for generic images
+to ensure that they can consistently compress into small but usable files.
+
+The pipeline in [scaler.ps1] is currently configured to:
+
+- Emit AVIF at quality 70 with YUV 4:4:4. For most scenes, this generally does well.
+- Emit JPEG at distance 1.5 with YUV 4:4:4. For nearly all scenes, this does well.
+- Emit 360p, 720p and 1080p scales of input.
+  - This allows for data saving and responsiveness as the client doesn't need to download as large of files.
```
Comment (Ampflower) [inline pictures.md:31]: This is a lie; this should be AVIF q60 and JPEG d2
720p was also omitted as it wasn't deemed to be worth while; while 4K is emitted.

File: pictures.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+With these quality considerations, for noisy shader-intensive images,
+the recommended encoding settings are of the following:
+
+- AVIF quality 60-80 with YUV 4:4:4. Highly scene dependent.
+- JPEG using Jpegli on distance 2-1.5 with YUV 4:4:4. Hardly changes with scene complexity.
+  - JPEG XL (a.k.a., JXL) should generally perform similary but with marginally smaller file sizes.
```
Comment (Ampflower) [inline pictures.md:38]: It is notable that cjxl emits higher quality output at the same given distance when compared to jpegli.

File: pictures.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+- Heavy visual degradation. Drop the format if it cannot meet expected quality under 200K at 1080p.
+- Any unexpected smoothing of lines
+- Any unexpected blurring between strong colours (i.e. reddish-pink & black)
+- Any strong colour blending that shouldn't exist
+- Any muteness in colour.
+
+## The encoders in use are:
+
+- libwebp's `cwebp -preset photo -m 6 -z 9 input.png output.webp`
+- libjxl's `cjxl input.png output.jxl "--brotli_effort=11" -e 9`
```
Comment (Ampflower) [inline pictures.md:19]: Should be `-e 10`

File: pictures.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+- Emit 360p, 720p and 1080p scales of input.
+  - This allows for data saving and responsiveness as the client doesn't need to download as large of files.
+
+With these quality considerations, for noisy shader-intensive images,
+the recommended encoding settings are of the following:
+
+- AVIF quality 60-80 with YUV 4:4:4. Highly scene dependent.
+- JPEG using Jpegli on distance 2-1.5 with YUV 4:4:4. Hardly changes with scene complexity.
+  - JPEG XL (a.k.a., JXL) should generally perform similary but with marginally smaller file sizes.
+- WebP quality 90 using photo mode.
```
Comment (Ampflower) [inline pictures.md:39]: I have noticed that drawing mode sometimes produces better images. Need to experiment more.

File: pictures.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  - Important: WebP 90 may not always be within quality constraints and thus is disqualified.
+  - Note: This is often larger than Jpegli.
+
+For flat images, the recommended settings are of the following:
+
+- AVIF quality 50-60 with YUV 4:4:4 or YUV 4:2:0.
+- JPEG using Jpegli on distance 1.5-1 with YUV 4:4:4.
+  - Note: This may be larger than WebP, and introduce more noise than favourable.
+- WebP quality 90 using photo mode.
```
Comment (Ampflower) [inline pictures.md:48]: Experimentation with drawing mode needed.

Although we have dropped WebP due to AVIF being baseline so likely irrelevant now.

File: pictures.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+  - Being a video codec, this ends up excelling over JPEG.
+
+### Encoders to avoid:
+
+- `libjpeg-turbo`, `mozjpeg`
+  - Offers subpar quality at a given size in comparison to `jpegli`.
```
Comment (Ampflower) [inline pictures.md:54]: Did you know: `jpegli` at -d 25 produces better quality and smaller jpegs than `libjpeg-turbo` at quality 0

File: pictures.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+- Non-AOM AVIF encoders, including the GNU Image Manipulation Program
+  - Fails to return the same visual quality at a given size; may introduce artifacts.
+- `cwebp` at quality <80
+  - Always fails in retaining visual quality even in simple scenes.
+
+## Encoded Quality Settings Picked
```
Comment (Ampflower) [inline pictures.md:60]: This section probably should be renamed as this isn't relevant to the current pictures.

File: src/lib/components/YouTube.svelte [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+	export let title: string;
+</script>
+
+<a href="https://youtu.be/{id}" class="video" {title}>
+	<!-- They took the gay away... -->
+	<img src="/assets/youtube-noiframe.avif" alt={title} hidden loading="lazy" />
+	<iframe
+		src="https://www.youtube-nocookie.com/embed/{id}?controls=0"
+		{title}
+		allow="clipboard-write; encrypted-media; picture-in-picture; web-share"
```
Comment (Ampflower) [inline src/lib/components/YouTube.svelte:12]: Probably make this bit configurable.

File: src/routes/frog/+page.svelte.md [hunk 13]
```diff
@@ -1,3 +1,8 @@
+<script lang="ts">
+let value = Math.floor(Math.random() * 999) + 1;
+</script>
+
 # You have been frogged!
-![gay frog is not real and cant hurt you. gay frog:](https://github.com/Pridecraft-Studios/website/blob/main/static/assets/img/frog/frog.png?raw=true)
-***420M*** *other people were also frogged!*
\ No newline at end of file
+
+![gay frog is not real and cant hurt you. gay frog:](/assets/frog.webp)
```
Comment (Ampflower) [inline src/routes/frog/+page.svelte.md:7]: Consider maybe an AVIF version of the frog.

**Review by Ampflower — COMMENTED** [review]
More things I've noticed

**Comment by Astralchroma** [comment]
image

**Comment by Ampflower** [comment]
Note: Permission to license already obtained from @blryface and @vgskye in private.

**Comment by blryface** [comment]
time to review ~~while *totally* being mentally able to do stuff correctly~~

**Review by blryface — APPROVED** [review]
LGTM!

**Review by KTrain5169 — CHANGES_REQUESTED** [review]
Not exactly bugs but figure I might as well mention them

image
Ideally there should be a tiny bit more space between dev list & footer

image
`worldwidepixel` is too small, didn't realise what it said at first

**Comment by blryface** [comment]
> Not exactly bugs but figure I might as well mention them
> 
> image Ideally there should be a tiny bit more space between dev list & footer
> 

Sure, that's true. We can just add a new /n there probably

> image `worldwidepixel` is too small, didn't realise what it said at first

That's fine it's just the copyright notice, the whole thing is smaller than the whole description

**Comment by Laxystem** [comment]
image
it doesn't work.

**Comment by Ampflower** [comment]
I noticed Stackblitz was broken too; I frankly don't get why either. Might just remove the bot given its not terribly useful.

**Comment by Ampflower** [comment]
Resolved the footer touching the developer section, ~~totally not by redoing the entire thing, no-~~ :3

**Comment by worldwidepixel** [comment]
TLDR: Everything to do with dev bios looks wrong in light mode.

**Review by KTrain5169 — APPROVED** [review]
Personally there's way too much in the `Developers` section that I think could be worth a subpage but that's like a really really *really* minor nitpick and also it looks cool and workable in the current state anyways so LGTM!

**Comment by Ampflower** [comment]
> TLDR: Everything to do with dev bios looks wrong in light mode.

Fixed; forced to dark mode for now, but considering making a light theme version.

**Comment by blryface** [comment]
It's sort of funny that this rewrite has taken over a month, *and all I asked for was to update the descriptions of projects*

It's not a bad thing, I love how the site is looking now

**Comment by blryface** [comment]
[context](https://discord.com/channels/1091969030694375444/1091977141815627946/1296276388076327004)

File: src/routes/pridegui/+page.svelte.md [hunk 14]
```diff
@@ -1,44 +1,48 @@
...
-Currently, there is no support planned for PrideGUI just yet, but mod support may be added in the future, after the initial project is finished.
+<Who/>
 
+## Where can I download PrideGUI?
 
-## Where is this avaliable?
+<ul class="badges">
+<li><Badge id="modrinth" rel="me" link="https://modrinth.com/resourcepack/pridegui" ext="svg" head="Available on" name="Modrinth"/></li>
+<li><Badge id="github" rel="me" link="https://github.com/Pridecraft-Studios/pridegui" ext="svg" head="Available on" name="GitHub"/></li>
+</ul>
```
Comment (blryface) [inline src/routes/pridegui/+page.svelte.md:42]: image
PrideGUI should have download links to all different editions (Normal, Legacy, Light Mode, Light Mode Legacy).

You can use less space by using the smaller badge design, like we do on Modrinth. On the website it could be separated in 4 collumns if you can figure it out

image

**Review by blryface — CHANGES_REQUESTED** [review]
otherwise lgtm

File: src/routes/pridegui/+page.svelte.md (hunk 14 above)
Comment (Ampflower) [reply src/routes/pridegui/+page.svelte.md:42]: It already shrinks for smaller views automatically; although I'd probably going to want to make that explicit; hm.

I'll figure something out; it'll probably be 4 columns that collapse to fit to the screen.

**Comment by KTrain5169** [comment]
Still at school unfortunately, wish they didn't block GitHub :(

File: src/routes/pridegui/+page.svelte.md (hunk 14 above)
Comment (blryface) [reply src/routes/pridegui/+page.svelte.md:42]: I meant you could use the smaller ones by default (on non W    I    D     E screens) to take up less space. Quite neat that it shrinks automatically.

Speaking of smaller screens, I should test it on mobile

Comment (blryface) [reply src/routes/pridegui/+page.svelte.md:42]: Screenshot_20241016-231335
Wasn't it meant to shrink?

Comment (Ampflower) [reply src/routes/pridegui/+page.svelte.md:42]: Hm, possible your display is considered too wide for the CSS to trip on that.

**Review by KTrain5169 — APPROVED** [review]

File: src/routes/faithfulpride/+page.svelte.md (hunk 3 above)
Comment (Ampflower) [reply src/routes/faithfulpride/+page.svelte.md]: Considering this out of scope.

**Comment by Ampflower** [comment]
I believe this should be complete, should no issues or need to make changes crop up.

> Personally there's way too much in the `Developers` section that I think could be worth a subpage but that's like a really really _really_ minor nitpick and also it looks cool and workable in the current state anyways so LGTM!

I should note: The developers section being on the home page was originally due to various devs wanting to link to Pridecraft on their fediverse accounts have it be considered verified.

It could do with being their own page, but frankly, as it is, it should be fine.

**Review by blryface — APPROVED** [review]
one new file, i'm requesting it to be changed to 100 new changes /s
lgtm

**Comment by blryface** [comment]
I also saw the pridegui CSS thing on discord, that looks good too. I don't know much webdev, but visually it seems to be finally done. Looks good to merge, maybe we should do a normal one instead of squash this time?

**Comment by blryface** [comment]
it's funny @worldwidepixel "didn't" approve it yet

**Comment by Ampflower** [comment]
Okay let's reopen the PR and have it be approved :3
Tho saying that I wonder if you can do a code review post-merge
